source/particle/interpolator/bilinear_least_squares.cc

```diff
@@ -0,0 +1,154 @@
+        for (unsigned int i=0; i < n_particle_properties; i++)
+          if (selected_properties[i])
+            property_index = i;
+
+        AssertThrow(n_particles != 0,
+                    ExcMessage("At least one cell contained no particles. The 'bilinear'"
+                               "interpolation scheme does not support this case. "));
+
+
+        const unsigned int matrix_dimension = 4;
+        dealii::LAPACKFullMatrix<double> A(n_particles, matrix_dimension);
+        Vector<double> r(n_particles);
+        A = 0;
+        r = 0;
+
+        unsigned int index = 0;
+        double cell_diameter = found_cell->diameter();
+        for (typename std::multimap<types::LevelInd, Particle<dim> >::const_iterator particle = particle_range.first;
+             particle != particle_range.second; ++particle, ++index)
+          {
+            const double particle_property_value = particle->second.get_properties()[property_index];
+            r[index] = particle_property_value;
+
+            const Point<dim> position = particle->second.get_location();
+            A(index,0) = 1;
+            A(index,1) = (position[0] - approximated_cell_midpoint[0])/cell_diameter;
+            A(index,2) = (position[1] - approximated_cell_midpoint[1])/cell_diameter;
+            A(index,3) = (position[0] - approximated_cell_midpoint[0]) * (position[1] - approximated_cell_midpoint[1])/std::pow(cell_diameter,2);
+          }
+
+        dealii::LAPACKFullMatrix<double> B(matrix_dimension, matrix_dimension);
+        dealii::LAPACKFullMatrix<double> B_inverse(B);
+
+        Vector<double> c_ATr(matrix_dimension);
+        Vector<double> c(matrix_dimension);
+
+        const double threshold = 1e-15;
+        unsigned int index_positions = 0;
+
+        A.Tmmult(B, A, false);
+        A.Tvmult(c_ATr,r);
+        B_inverse.compute_svd();
+        B_inverse.compute_inverse_svd(threshold);
+        B_inverse.vmult(c, c_ATr);
+
+        for (typename std::vector<Point<dim> >::const_iterator itr = positions.begin(); itr != positions.end(); ++itr, ++index_positions)
+          {
+            Point<dim> support_point = *itr;
+            double interpolated_value = c[0] +
+                                        c[1]*(support_point[0] - approximated_cell_midpoint[0])/cell_diameter +
+                                        c[2]*(support_point[1] - approximated_cell_midpoint[1])/cell_diameter +
+                                        c[3]*(support_point[0] - approximated_cell_midpoint[0])*(support_point[1] - approximated_cell_midpoint[1])/std::pow(cell_diameter,2);
+            cell_properties[index_positions][property_index] = interpolated_value;
+          }
+        return cell_properties;
+      }
+    }
+  }
+}
+
+
+// explicit instantiations
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Interpolator
+    {
+      ASPECT_REGISTER_PARTICLE_INTERPOLATOR(BilinearLeastSquares,
+                                            "bilinear",
```

Comment:
would you mind changing the input file name to "bilinear least squares"? Somebody might have the idea to add a "bilinear nearest neighbor" or something similar later on.